File: Source/Rewards/OldSaves/OldsaveSync.cs
using WCSharp.Api;
using static WCSharp.Api.Common;
public static class OldsaveSync
{
    private const string SYNC_PREFIX = "S";
    private static trigger Trigger = trigger.Create();
    private static trigger VariableEvent = trigger.Create();
    private static string SaveLoadCode { get; set; }
    private static player SavePlayer { get; set; }
    public static float LoadEvent { get; set; }

    public static bool SyncString(string s)
    {
        return BlzSendSyncData(SYNC_PREFIX, s);
    }

    public static void Initialize()
    {
        foreach (var player in Globals.ALL_PLAYERS)
        {
            BlzTriggerRegisterPlayerSyncEvent(Trigger, player, SYNC_PREFIX, false);
        }
        Trigger.AddAction(() =>
        {
            SavePlayer = GetTriggerPlayer();
            SaveLoadCode = BlzGetTriggerSyncData();
            LoadActions();
        });
    }

    public static void LoadActions()
    {
        Savecode savecode = new Savecode();
        if (SaveLoadCode.Length < 1) return;
        if (!savecode.Load(SavePlayer, SaveLoadCode))
        {
            SavePlayer.DisplayTimedTextTo(5.0f, $"{Colors.COLOR_RED}The save code is invalid :(");
            return;
        }
        savecode.SetRewardValues(SavePlayer);
        MultiboardUtil.RefreshMultiboards();

    }
}

File: src/Rewards/OldSaves/OldsaveSync.ts
import { Globals } from 'src/Global/Globals'
import { MultiboardUtil } from 'src/UI/Multiboard/MultiboardUtil'
import { getTriggerPlayer } from 'src/Utility/w3tsUtils'
import { MapPlayer, Trigger } from 'w3ts'
import { Savecode } from './OldSaves'

export class OldsaveSync {
    private static SYNC_PREFIX: string = 'S'
    private static Trigger: Trigger = Trigger.create()!
    private static VariableEvent: Trigger = Trigger.create()!
    private static SaveLoadCode: string
    private static SavePlayer: MapPlayer
    public static LoadEvent: number

    public static SyncString(s: string) {
        return BlzSendSyncData(OldsaveSync.SYNC_PREFIX, s)
    }

    public static Initialize() {
        for (let player of Globals.ALL_PLAYERS) {
            OldsaveSync.Trigger.registerPlayerSyncEvent(player, OldsaveSync.SYNC_PREFIX, false)
        }
        OldsaveSync.Trigger.addAction(() => {
            OldsaveSync.SavePlayer = getTriggerPlayer()
            OldsaveSync.SaveLoadCode = BlzGetTriggerSyncData() || ''
            OldsaveSync.LoadActions()
        })
    }

    public static LoadActions() {
        let savecode: Savecode = new Savecode()
        if (OldsaveSync.SaveLoadCode.length < 1) return
        if (!savecode.Load(OldsaveSync.SavePlayer, OldsaveSync.SaveLoadCode)) {
            OldsaveSync.SavePlayer.DisplayTimedTextTo(5.0, '{Colors.COLOR_RED}save: code: The is invalid :(')
            return
        }
        savecode.SetRewardValues(OldsaveSync.SavePlayer)
        MultiboardUtil.RefreshMultiboards()
    }
}
